R/guides-.R

```diff
@@ -502,43 +563,47 @@ Guides <- ggproto(
 
     Map(
       grobs    = grobs,
-      position = names(grobs),
+      position = positions,
+      inside_position = inside_positions,
+      inside_just = inside_justs,
       self$package_box,
       MoreArgs = list(theme = theme)
     )
   },
 
   # Render the guides into grobs
-  draw = function(self, theme,
-                  default_position = "right",
-                  direction = NULL,
+  draw = function(self, theme, positions = NULL, direction = NULL,
                   params = self$params,
                   guides = self$guides) {
-    positions <- vapply(
+    positions <- positions %||% vapply(
       params,
-      function(p) p$position[1] %||% default_position,
-      character(1)
+      function(p) p$position[1] %||% "right",
+      character(1), USE.NAMES = FALSE
     )
```

Comment:
Can't we assume here that the `Guides$draw()` method receives a well-formed `positions` argument, so we don't have to fall back here? You could even change `positions = NULL` to `positions` in the arguments to signal that it is a required argument.